test/es-module/test-cjs-legacyMainResolve-permission.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+// Flags: --expose-internals --experimental-permission --allow-fs-read=* --allow-child-process
+
+require('../common');
+
+const { describe, it } = require('node:test');
+const assert = require('node:assert');
+const fixtures = require('../common/fixtures.js');
+
+const { spawnSync } = require('node:child_process');
+
+describe('legacyMainResolve', () => {
+  it('should ensure permission model when resolving by packageConfig.main', () => {
+    const paths = [
+      './index-js/index.js',
+      './index-js/index',
+      './index-json/index',
+      './index-node/index',
+      './index-js',
+      './index-json',
+      './index-node',
+    ];
+
+    const fixtextureFolder = fixtures.path('/es-modules/legacy-main-resolver');
+
+    for (const main of paths) {
+      const { status, stderr } = spawnSync(
+        process.execPath,
+        [
+          '--expose-internals',
+          '--experimental-permission',
+          '-e',
+          `
+            const { legacyMainResolve } = require('node:internal/modules/esm/resolve');
+            const { pathToFileURL } = require('node:url');
+            const path = require('node:path');
+            const assert = require('node:assert');
+
+            const packageJsonUrl = pathToFileURL(
+              path.resolve(
+                '${fixtextureFolder}',
+                'package.json'
+              )
+            );
+
+            const packageConfig = { main: '${main}' };
+            const base = path.resolve(
+              '${fixtextureFolder}'
+            );
+
+            assert.throws(() => legacyMainResolve(packageJsonUrl, packageConfig, base), {
+              code: 'ERR_ACCESS_DENIED',
```

Comment:
I could not make the test run with the permission model, I sent you a message on Slack with the details since I don't know if it is a bug on the permission model.

Reply:
It was indeed a bug, a PR was created to fix it, waiting for https://github.com/nodejs/node/pull/48531 to get back to work on testing for the permission model.